### ISSUE TITLE
taking "screen captures" to the next level! 

### ISSUE THREAD

**Issue by zldrq** [deanoemcke/thegreatsuspender#217]
i really think there must be an option to make screen captures of a whole page (with scroll)

the motivation is:
nowadays memory is not much of an issue. actually me (and i think many) use TGS to disable scripts, media and all the likes running on pages to save CPU, rather than to save RAM (which is nice addition too).

so it will be great to see the option that captures the whole page 
like for example this extension does https://chrome.google.com/webstore/detail/full-page-screen-capture/fdpohaocaechififmbbbbbknoalclacl?utm_source=chrome-app-launcher-info-dialog

if the page is TOO tall, may be save the image just 4000 pixels height or smth

and leave options to choose quality. so actually it will be 4 possible combinations
low res - single page
hi res - single page
low res - tall page
hi res - tall page

**Comment by yadomi** [comment]
:+1:

**Comment by deanoemcke** [comment]
@zldrq this is a fair request. although i would question your statement that memory is not an issue. i'm fairly confident that the majority of users of this extension use it to solve their memory problems, not their cpu load. i'll consider this for a future release.

**Comment by Dionio** [comment]
Yes, it would be nice feature! I came here to create the same feature request.

My use case is simple - I open a page with information and suspend the tab. Later I use suspended tab as a source of useful information (is case of stackoverflow, msdn, etc). I open a lot of such pages during solving tech problems, I guess, you too.
If a whole page is captured I can view all the required information without unsuspending it. Currently I have to unsuspend it to view the content below the end of first page.
Option to limit capture vertical size is nice too.

About memory usage - in my Chrome 15+ suspended tabs with high-res captures takes 85Mb, which is less than one modern webpage, so I think, using tall high-res captures will still save memory very good. As a plus it will be possible to avoid unsuspending pages to view hidden content.

More options are (I mean, for the far future):
1. Overlay with plain text content of the page, without styles and/or markup, to be able copy some text from page (code snippets, links, etc).
2. White list for "Always full page capture" for StackOverflow, MSDN, etc
3. Integration with one of "clean-read" extensions, who removes all unrelated information from screen.

And HUGE thank you for The Plugin!

**Comment by zldrq** [comment]
@Dionio 
"open a lot of such pages during solving tech problems, I guess, you too."
exactly what i need it for!

"overlay with plain text" seems a bit of overkill tho (=

@deanoemcke yes, memory is an issue on low-memory machines (4gb, may be even 8), that's why i propose making this feature an option, ofc
it's nice that 150+ suspended tabs take only 250mb, but i'd sacrifice some memory for them to be instantly readable
and nice suggestion from @Dionio indeed: white list. thinking of it, i need just some pages available (SO, MSDN, yes), others (like youtube, social media, news) may rest until i deal with them in spare time and close

**Comment by deanoemcke** [comment]
I have changed the way screen captures work. I have removed the low-quality option - I did some tests and the memory savings isn't significant enough to justify having bad images.
I have added two new options - capture visible screen, and capture whole page. the latter will give you a scroll bar as requested.
I am toying with the idea of some kind of offline mode that would allow copying of text - but will address this in a seperate issue if I do implement it.

**Comment by Dionio** [comment]
Hurray! :) Thank you!
>> I am toying with the idea of some kind of offline mode that would allow copying of text
Yes, it would be nice feature! Actually, in many cases (news pages, blogposts to read, tech threads, etc) user will not interact with web-pages after rendering, except scrolling and text searching, or may be clicking on one-two links on it. So browser in many cases is required only for loading+rendering web-pages, and after that web-page can be converted into full render of page to save memory. That do you think about this?

**Comment by zldrq** [comment]
wow so cool. when will it be live on webstore?

**Comment by Dionio** [comment]
> I have changed the way screen captures work.
Please, tell me, that did you mean? Did you changed capturing technique from html rendering to smthng? Actually, js-renderer produces images with artifacts, not all html tags or CSS styles are supported, so may be it is possible to try to get series of raw screenshots using scroll, and after that combine them in one image? I know, several Chrome add-ons for screenshot-ing are working this way.

**Comment by zldrq** [comment]
@Dionio actually I think it's ok in current state. Almost all pages are readable, tho more often than not don't have images. But perfect capture is a bit out of scope of this plugin. If, of course, author wishes to expand it (as he does, pretty much) to be something bigger.
Anyway, I don't think (i'm sure, actually) the method you proposing is suitable for capturing tabs in background and also is really slow, so the best strategy here will be to monitor rendering engines and hook up the best. For time being extension may be renamed to Greater Suspender =))

tl;dr being auxiliary feature, it is decent, but when it begin render (almost) as good as browsers, this extension will become almost something else and surely something bigger. Top 5 valuable on webstore. The name will have to be changed to Greatest Suspender, apparently

**Comment by IanEdington** [comment]
@deanoemcke :+1: 

Came here to make this feature request. Pleasantly surprised to see you already did it :+1: :+1: :+1:  :smile: and for good measure :tada:

**Comment by IanEdington** [comment]
@deanoemcke I've been using the feature-full-screen-captures branch for the last week and it's working well. I want to echo @zldrq. The addition of an html/css dump that leaves the text selectable and links clickable without the overhead of a dynamic page would make this the "Greatest Suspender".

**Comment by zldrq** [comment]
@IanEdington Actually, I didn't say anything like that. I think feature with text, links etc is a bit overkill. The game is not worth the candle as a least. 
One could state, having tab no closed, means you wanna refer to it later. In this case having visual representation is a great thing, as you can quickly remember, what's on page and close it, read it or sort. The necessity of searching its text or clicking links? Click on page, load page. If you can't load page, when you are offline, having clickable links won't help you much. Having text? Pretty much unlikely, since searching text on a page as action is pretty much always comes with surfing..
If you know you gonna need a page, press ctrl+p, save as pdf..

As for feature itself. after using it for some time, I think it's a huge win! My workflow had improved drastically! Being able to quickly parse page (not only top part with article title) is great. Now I can more easily organize 150-250 tabs in several windows and i see not the lists of page titles, but pictures!

**Comment by synnack** [comment]
There seems to be a problem in the master branch with this feature, the scrollbars don't show in chromium 48 64-bit on Linux..

If in suspended.js, I change line 82 from:
                    document.body.style['overflow-x'] = 'auto';
to:
                    document.body.style['overflow-y'] = 'auto';

It does work. Perhaps it's a good idea to allow x and y scrolling?

**Comment by deanoemcke** [comment]
Now available in the chrome webstore in v6.30

**Comment by Dionio** [comment]
This is excellent news! Thank you!!
